### DISCUSSION TITLE
How to set default runtime (for example, nvidia) for k3s?

### DISCUSSION THREAD

**Discussion by npuichigo** [k3s-io/k3s#9705] [Q&A]
https://github.com/k3s-io/k3s/blob/fe2ca9ecf1cbf631b22dc5d5acd923a1db113b9d/pkg/agent/templates/templates_linux.go#L39

Where does these values like `.NodeConfig.DefaultRuntime` and  `.NodeConfig.DefaultRuntime` come from? Since I don't want to always specify `runtimeClassName: nvidia`, is there any way to specify that on installation?

**Comment by brandond** [comment]
`   --default-runtime value                    (agent/runtime) Set the default runtime in containerd`

I don't believe the CLI docs on the website have been updated since we added this option, but it does show up in `k3s server --help` and `k3s agent --help`

**Reply by npuichigo** [reply]
If I don’t want to uninstall and reinstall my k3s cluster. Is there anyway to reconfigure the cluster to change default runtime?

**Reply by DavraYoung** [reply]
@npuichigo just edit the unit file on all nodes:

For server:
` nano /etc/systemd/system/k3s.service`
```
....
ExecStart=/usr/local/bin/k3s \
    server --default-runtime nvidia\
```


For agent:
` nano /etc/systemd/system/k3s-agent.service`
```
....
ExecStart=/usr/local/bin/k3s \
    agent --default-runtime nvidia\
```

**Reply by SezBulent** [reply]
FYI, after all this time the docs still haven't been updated to include this arg.

**Reply by brandond** [reply]
PR welcome @SezBulent !
https://github.com/k3s-io/docs
